### PR TITLE
shfmt 1.2.0 (new formula)

### PR DESCRIPTION
**Pull request by deiga**
Adds formula to download and build shfmt with GNU Go

### CODE REVIEW

File: Formula/shfmt.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class Shfmt < Formula
+  desc "Autoformat shell script source code"
+  homepage "https://github.com/mvdan/sh"
+  url "https://github.com/mvdan/sh/archive/v1.2.0.tar.gz"
+  sha256 "3d2973f1adf99fcf65baae3c85697313a782dbedc2600fedb28687541a20ed43"
+  head "https://github.com/mvdan/sh.git"
+  depends_on "go" => :build
+
+  def install
+    version = "1.2.0"
```
Comment (bfontaine) [inline Formula/shfmt.rb]: You can remove this line, the `version` is already guessed from the formula URL.

File: Formula/shfmt.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    ENV["GOPATH"] = buildpath/".."
```
Comment (bfontaine) [inline Formula/shfmt.rb]: Don’t try to do stuff outside of the `buildpath`.

File: Formula/shfmt.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    mkdir_p buildpath/"../src/github.com/mvdan"
```
Comment (bfontaine) [inline Formula/shfmt.rb]: Use `(buildpath/"foo").mkpath` instead of `mkdir_p buildpath/"foo"`.

File: Formula/shfmt.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    mv buildpath/"../sh-#{version}", buildpath/"../src/github.com/mvdan/sh"
```
Comment (bfontaine) [inline Formula/shfmt.rb]: Use `(buildpath/"../src/github.com/mvdan/sh").install "../sh-#{version}"` instead (with the correct path).

File: Formula/shfmt.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    mkdir buildpath/"sh-#{version}" # so cleanup doesn't fail
```
Comment (bfontaine) [inline Formula/shfmt.rb]: Again, use `.mkpath` instead of `mkdir`.

File: Formula/shfmt.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    cd buildpath/".." do
+      system "go", "build", "-a", "-tags", "production brew", "github.com/mvdan/sh/cmd/shfmt"
+      bin.install "shfmt"
```
Comment (bfontaine) [inline Formula/shfmt.rb]: Use `system "go", "build", ..., "-o", "#{bin}/shfmt"` to avoid this line.

File: Formula/shfmt.rb [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    end
+  end
+
+  test do
+    touch "shfmt-test-empty"
```
Comment (bfontaine) [inline Formula/shfmt.rb]: You can create a minimal test file with e.g. `(testpath/"test").write "echo foo"`.

File: Formula/shfmt.rb [hunk 8]
```diff
@@ -0,0 +1,25 @@
+class Shfmt < Formula
+  desc "Autoformat shell script source code"
+  homepage "https://github.com/mvdan/sh"
+  url "https://github.com/mvdan/sh/archive/v1.2.0.tar.gz"
+  sha256 "3d2973f1adf99fcf65baae3c85697313a782dbedc2600fedb28687541a20ed43"
+  head "https://github.com/mvdan/sh.git"
+  depends_on "go" => :build
```
Comment (bfontaine) [inline Formula/shfmt.rb]: Please leave one blank line above that one.

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/shfmt.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  head "https://github.com/mvdan/sh.git"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/mvdan").mkpath
+    (buildpath/"src/github.com/mvdan/sh").install buildpath/"cmd"
+    (buildpath/"src/github.com/mvdan/sh").install buildpath/"fileutil"
+    (buildpath/"src/github.com/mvdan/sh").install buildpath/"syntax"
```
Comment (bfontaine) [inline Formula/shfmt.rb]: You should be able to merge all lines above except the `ENV` one into that one:
```ruby
(buildpath/"src/github.com/mvdan/sh").install "*"
```

`.install` automatically create the missing directories.

Comment (deiga) [inline Formula/shfmt.rb]: @bfontaine `.install` can't locate anything with `"*"`

Comment (bfontaine) [inline Formula/shfmt.rb]: Sorry I meant `Dir["*"]`.

**Comment by mvdan** [comment]
Also see @focusauru's existing formula

**Comment by deiga** [comment]
@bfontaine Everything should be good now.
@mvdan I saw and decided that this is popular enough to be inside homebrew-core

This way it also get's some better maintenance

**Review by bfontaine — APPROVED** [review]

**Comment by focusaurus** [comment]
Would be great to get this into homebrew core! FYI I've updated my tap to v1.2.0.

**Comment by fxcoudert** [comment]
Thanks @deiga for your contribution!

**Pull request closed without merging**